src/replication.c

```diff
@@ -3677,7 +3680,8 @@ void processClientsWaitingReplicas(void) {
             addReplyLongLong(c, numlocal);
             addReplyLongLong(c, numreplicas);
         } else {
-            addReplyLongLong(c, numreplicas);
+            /* Don't reply yet if there is a pending command */
+            if (!(c->flags & CLIENT_PENDING_COMMAND)) addReplyLongLong(c,numreplicas);
```

Comment:
i don't understand this change.
we're gonna unblock the client without sending a reply?

Reply:
Good catch @oranagra. This change is no longer needed and should be removed, now that the wait is done outside of the engine and this check will always return true so a reply is guaranteed. Previously, CLIENT_PENDING_COMMND can happen on this path if the engine itself decides to hold off the slot finalization on the primary until it is replicated.

Reply:
change reverted